src/contexts/meetupContext.tsx

```diff
@@ -0,0 +1,646 @@
+    throw new Error("useMeetupDispatch must be used within a MeetupProvider");
+  }
+  return context as Dispatch;
+}
+
+function useMeetup(): [MeetupState, Dispatch] {
+  return [useMeetupState(), useMeetupDispatch()];
+}
+
+async function getMeetupDetails(token: string, meetupID: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting?id=${meetupID}&token=${token}`;
+  return await AuthenticatedRequestHandler.get(endpoint, mapMeetupResponseJson);
+}
+
+async function getMeetupList(token: string, email: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting/${email}?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.get(
+    endpoint,
+    mapMeetupListResponseJson
+  );
+}
+
+async function getPendingMeetupsList(token: string, email: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting/pending?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.get(
+    endpoint,
+    mapMeetupListResponseJson
+  );
+}
+
+async function getUpcomingMeetupsList(token: string, email: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting/upcoming?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.get(
+    endpoint,
+    mapMeetupListResponseJson
+  );
+}
+
+async function getUnratedMeetupsList(token: string, email: string) {
+  const endpoint = `${SERVER_ADDRESS}/rating/pending?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.get(
+    endpoint,
+    mapRatingListResponseJson
+  );
+}
+
+async function setMeetupRating(
+  token: string,
+  email: string,
+  meetupID: string,
+  rating: number
+) {
+  const endpoint = `${SERVER_ADDRESS}/rating`;
+  return await AuthenticatedRequestHandler.post(
+    endpoint,
+    mapRatingResponseJson,
+    { token, user: email, meeting_id: meetupID, rating } // eslint-disable-line camelcase
+  );
+}
+
+async function acceptMeetup(token: string, email: string, meetupID: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting/${meetupID}/accept?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.put(endpoint, mapMeetupResponseJson);
+}
+
+async function declineMeetup(token: string, email: string, meetupID: string) {
+  const endpoint = `${SERVER_ADDRESS}/meeting/${meetupID}/decline?email=${email}&token=${token}`;
+  return await AuthenticatedRequestHandler.put(endpoint, mapMeetupResponseJson);
+}
+
+async function setMeetupLocations(
+  token: string,
+  email: string,
+  meetupID: string,
+  venues: string[]
+) {
+  let venuesAsString = "[";
+  for (let index = 0; index < venues.length; index++) {
+    venuesAsString += `"${venues[index]}",`;
+  }
+  venuesAsString = venuesAsString.substring(0, venuesAsString.length - 1);
+  venuesAsString += "]";
+  console.log(venuesAsString);
+  const endpoint = `${SERVER_ADDRESS}/meeting/${meetupID}/venue?email=${email}&token=${token}&venues=${venuesAsString}`;
+  return await AuthenticatedRequestHandler.put(endpoint, mapMeetupResponseJson);
+}
```

Comment:
theses functions look very good, but  you aren't using the dispatch function to update state, this is gonna create a few errors I think, I can help you debug this at the meeting today.

Reply:
That sounds good. In my testing I haven't ran into any errors, but also technically these aren't relying on a global state, rather we call and get server updates when loading the screen